src/inference/src/dev/core_impl.cpp

```diff
@@ -1473,21 +1486,46 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::load_model_from_cache(
                 update_config[ov::loaded_from_cache.name()] = true;
 
                 if (util::contains(plugin.get_property(ov::supported_properties), ov::weights_path)) {
-                    std::string weights_path = cacheContent.modelPath;
-                    auto pos = weights_path.rfind('.');
-                    if (pos != weights_path.npos && weights_path.substr(pos) == ".xml") {
-                        weights_path = weights_path.substr(0, pos);
-                        weights_path += ".bin";
+                    util::Path weights_path;
+
+                    if (auto&& path_hint = update_config.find(ov::weights_path.name());
+                        path_hint != update_config.end()) {
+                        weights_path = path_hint->second.as<std::string>();
+                    } else if (weights_path = extract_weight_path(header.get_runtime_info()); weights_path.empty()) {
+                        weights_path = cacheContent.modelPath;
+                        weights_path.replace_extension(".bin");
                     }
+
                     if (ov::util::file_exists(weights_path)) {
-                        update_config[ov::weights_path.name()] = weights_path;
+                        update_config[ov::weights_path.name()] = weights_path.string();
                     }
                 }
                 if (model_buffer) {
-                    update_config[ov::internal::cached_model_buffer.name()] = model_buffer;
+                    auto compiled_blob =
+                        Tensor(element::from<char>(), Shape{model_buffer->size()}, model_buffer->get_ptr());
+                    auto impl = get_tensor_impl(compiled_blob);
+                    impl._so = model_buffer;
+                    compiled_blob = make_tensor(impl);
```

Comment:
what a dirty hack... 
I like it  :D

Reply:
but does it works only if we don't share this tensor to another module?

Reply:
Tensor will keep the buffer alive until removed. Is working like tensor which holds object ownership.

What do you mean by "don't share it to another module"?

Reply:
I mean that common ov::Tensor has handle to the dll in **_so**. And now it contains custom shared obj in this field.
What if we create such tensor in one dll and then move it to another? In this case, it should be ok. But what about "mmaped tensor", where it can be reused

Reply:
The `_so` may store some shared object.
This tensor is created as view to some buffer (the _so is nullptr) and using dev API to set _so to mmaped object which will keep it alive until tensor end of live. There is no need to store any handle to shared library.

Reply:
w/o such hack, we can use custom allocator which will hold base reference.